### PR TITLE
Feat/in-store-review

### PR DESCRIPTION
**Pull request by dhruvkelawala**

### CODE REVIEW

**Comment by dhruvkelawala** [comment]
rebased on #1153

**Comment by simonheys** [comment]
> rebased on #1153

Great - maybe should this PR be changed to merge onto that same branch (`fix/notifications`)?

**Comment by dhruvkelawala** [comment]
> > rebased on #1153
> 
> Great - maybe should this PR be changed to merge onto that same branch (`fix/notifications`)?

My idea was to wait for your PR to be merged and then merge mine

File: packages/extension/src/ui/features/userReview/ReviewFeedbackScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
+import { FC } from "react"
+import { isFirefox } from "react-device-detect"
```
Comment (simonheys) [inline packages/extension/src/ui/features/userReview/ReviewFeedbackScreen.tsx:2]: Must avoid adding new dependencies, given the limited use-case at the moment can instead use something like?

`const isFirefox = navigator.userAgent.indexOf("Firefox") !== -1`

File: packages/extension/src/background/transactions/onupdate/notifications.ts [hunk 2]
```diff
@@ -16,6 +17,8 @@ export const notifyAboutCompletedTransactions: TransactionUpdateListener =
       ) {
         addToAlreadyShown(hash)
 
+        await decrementTransactionsBeforeReview()
```
Comment (simonheys) [inline packages/extension/src/background/transactions/onupdate/notifications.ts:20]: Just for sanity should probably move inside the `if (!account.hidden) {` check below?

File: packages/extension/src/ui/features/accountTokens/AccountTokens.tsx [hunk 3]
```diff
@@ -70,11 +72,24 @@ export const AccountTokens: FC<AccountTokensProps> = ({ account }) => {
...
 
   const showPendingTransactions = pendingTransactions.length > 0
   const accountName = getAccountName(account, accountNames)
   const network = useCurrentNetwork()
 
+  useEffect(() => {
+    if (!userHasReviewed && transactionsBeforeReview === 0) {
+      setTimeout(() => navigate(routes.userReview()), 1000)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (simonheys) [inline packages/extension/src/ui/features/accountTokens/AccountTokens.tsx:90]: Can you add a short note about why this is disabled here?

**Comment by simonheys** [comment]
> e2e tests are failing on ci but it's passing locally. Can you guys please run them locally and check?

Working locally for me, I've just re-run the CI

**Review by simonheys — APPROVED** [review]

**Pull request merged**